map.go

```diff
@@ -1486,10 +1490,34 @@ func marshalMap(m *Map, length int) ([]byte, error) {
 	return buf, nil
 }
 
-// MapIterator iterates a Map.
+// MapIterator is the interface defining methods to iterate
+// through the content of the target map.
+//
+// See Map.Iterate.
+type MapIterator interface {
+	// Err returns any encountered error.
+	//
+	// The method must be called after Next returns nil.
+	//
+	// For key-value maps, returns ErrIterationAborted if
+	// it wasn't possible to do a full iteration.
+	Err() error
+
+	// Next decodes the next key and value.
+	//
+	// In case of a value-only map (Queue and Stack), the key
```

Comment:
This is an implementation detail, not something the caller needs to take into account explicitly to correctly use the interface. Semantically, `keyOut` doesn't seem different from regular kv maps, so I'd drop this paragraph.